crates/locutus-node/src/user_events.rs

```diff
@@ -64,6 +66,15 @@ pub(crate) mod test_utils {
             self.owned_contracts.extend(contracts.into_iter())
         }
 
+        /// Events that the user generate.
+        pub fn generate_events(&mut self, events: impl IntoIterator<Item = UserEvent>) {
+            self.events_to_gen.extend(events.into_iter())
+        }
+
+        fn gen_events(&mut self) -> UserEvent {
+            self.events_to_gen.get(0);
+        }
+
         fn gen_new_event(&mut self) -> UserEvent {
```

Comment:
we can change this to `generate_rand_event`

Reply:
ok, this sounds better